packages/mds-config/types.ts

```diff
@@ -13,10 +13,18 @@
     See the License for the specific language governing permissions and
     limitations under the License.
  */
-import { ApiRequest, ApiResponse } from '@mds-core/mds-api-server'
+import { ApiRequest, ApiResponse, ApiResponseLocals } from '@mds-core/mds-api-server'
 import { Params, ParamsDictionary } from 'express-serve-static-core'
 
 export type ConfigApiRequest<P extends Params = ParamsDictionary> = ApiRequest<P>
-export type ConfigApiResponse = ApiResponse
+export interface ConfigApiResponse extends ApiResponse {
+  locals: ApiResponseLocals & {
+    properties: string | string[]
+  }
+}
 
-export type ConfigApiGetSettingsRequest = ConfigApiRequest<{ name: string }>
+export type ConfigApiGetSettingsRequest = ConfigApiRequest<{ property: string }>
+
+export interface ConfigApiGetMergedSettingsRequest extends ConfigApiRequest {
+  query: Partial<{ [P in 'p']: string | string[] }>
```

Comment:
Can you explain what this line is doing? It's a little unclear to me.

Reply:
It is saying that there is a query string parameter named `p` that is optional and can also appear more that once. Equivalent would be:
```
query: {
  p?: string | string[]
}
```
The chosen syntax (used other places is well) is convenient because query string parameters are _always_ strings. The common usage is:
```
query: Partial<{ [P in 'a' | 'b' | 'c']: string }>
```